core/iwasm/common/wasm_runtime_common.c

```diff
@@ -1185,7 +1215,11 @@ wasm_runtime_load(uint8 *buf, uint32 size, char *error_buf,
     if (get_package_type(buf, size) == Wasm_Module_Bytecode) {
 #if WASM_ENABLE_INTERP != 0
         module_common =
-            (WASMModuleCommon *)wasm_load(buf, size, error_buf, error_buf_size);
+            (WASMModuleCommon *)wasm_load(buf, size,
+#if WASM_ENABLE_MULTI_MODULE != 0
+                                          true,
+#endif
+                                          error_buf, error_buf_size);
```

Comment:
My suggestion is
- call `set_error_buf()` when `module_common` is NULL (meaning `wasm_load` or `aot_load_from_aot_file` just failed) in `wasm_runtime_load()`
- call `set_error_buf()` when `module_common` is NULL (meaning `wasm_load_from_sections` or `aot_load_from_sections` just failed) in `wasm_runtime_load_from_sections()`

Reply:
I  used` LOG_DEG() `instead of `set_error_buf()`, Otherwise it will rewrite the original error during the `wasm_load` or `aot_load_from_aot_file` process.

Reply:
I see. 

Please run the *imports.wast* with #2550. There will be a loading failure without any error message. It turns out that the failure comes from `load_import_func_info()`. Please check which branch causes the trouble and `set_error_buf()` with proper information.

Reply:
Maybe I know this error. It because WAMR's running-mode is AOT but loads WASM type's sub-module.
I have fixed and not push yet.
But combined with your modifications on#2550, it seems that there are still some problems with the generation of submodule AOT file.

Reply:
Would u mind sharing us details of problems?

Reply:
I have pushed the code, Before merging your changes, there was only an error about the aot file not being found (because it was not generated).
I tried to merge your changes, but there were more failed cases. You can confirm it on this branch.


```
==================== LAST LOG of linking ====================

  cpu features:  
  opt level:     3
  size level:    3
  output format: AoT file

Starting interpreter for module '/tmp/tmp9gekqync.aot'
Running: ../../../product-mini/platforms/linux/build/iwasm --module-path=/tmp --heap-size=0 -v=0 --repl /tmp/tmp9gekqync.aot
THE FINAL EXCEPTION IS Failed:
  expected: 'OSError(5, 'Input/output error')'
  got: 'Read file to buffer failed: open file /tmp/Mf.aot failed.
AOT module load failed: unknown import
'
Compiling AOT to '/tmp/tmpui2p6_yv.aot'
Running: ../../../wamr-compiler/build/wamrc --target=x86_64 --cpu=skylake --disable-simd --format=llvmir-opt --disable-llvm-lto -o /tmp/tmpui2p6_yv.aot /tmp/tmpeuk4o6ks.wasm
Invalid file type: expected wasm file but got other
Leaving tempfiles: ['/tmp/tmp3jwp2zbp.wast', '/tmp/tmpeuk4o6ks.wasm']


==================== LAST LOG END ====================

```

```
==================== LAST LOG of imports ====================

  expected: 'OSError(5, 'Input/output error')'
  got: 'Read file to buffer failed: open file /tmp/test.aot failed.
AOT module load failed: unknown import
'
Compiling AOT to '/tmp/tmp4zugl09f.aot'
Running: ../../../wamr-compiler/build/wamrc --target=x86_64 --cpu=skylake --disable-simd --format=llvmir-opt --disable-llvm-lto -o /tmp/tmp4zugl09f.aot /tmp/tmp47dxp1l6.wasm
Create AoT compiler with:
  target:        x86_64
  target cpu:    skylake
  target triple: x86_64-pc-linux-gnu
  cpu features:  
  opt level:     3
  size level:    3
  output format: optimized LLVM IR
Compile success, file /tmp/tmp4zugl09f.aot was generated.
Leaving tempfiles: ['/tmp/tmp2b31w2x1.wast', '/tmp/tmp47dxp1l6.wasm']


==================== LAST LOG END ====================

```

Reply:
@lum1n0us, could you help check when you have time?